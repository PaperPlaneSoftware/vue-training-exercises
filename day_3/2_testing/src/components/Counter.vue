<template>
  <div class="counter-container">
    <button @click="dec">-</button>
    <span>{{ count }}</span>
    <button @click="inc">+</button>
  </div>
</template>

<script>
export default {
  name: 'Counter',
  props: { 
    initCount: {
      type: Number,
      default: 42
    }
  },
  data: function() {
    return {
      count: this.initCount
    }
  },
  methods: {
    inc: function() {
      this.count++;
    },
    dec: function() {
      this.count--
    }
  }
}
</script>

<style scoped>
  .counter-container {
    display: flex;
    width: 200px;
    justify-content: space-between;
    border: 1px solid black;
  }

  .counter-container > span {
    width: 100px;
    text-align: center;
    font-size: 18pt;
    font-weight: bold;
  }
</style>
