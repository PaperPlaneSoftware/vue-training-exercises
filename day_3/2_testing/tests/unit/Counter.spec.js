import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter.vue'

// mounting a component and testing it's methods and internal state
describe('Counter.vue', () => {
  it('Should initialise to 42 by default', () => {
    const wrapper = shallowMount(Counter, {
      propsData: { }
    });
    expect(wrapper.vm.count).toEqual(42);
  });

  it("Should increment count by one", () => {
    const wrapper = shallowMount(Counter, {
      propsData: { initCount: 0 }
    });

    wrapper.vm.inc();
    expect(wrapper.vm.count).toEqual(1);
  });
});

// snapshot testing
describe('Counter.vue snapshot testing', () => {
  it("Snapshots should remain unchanged", () => {
    const wrapper = shallowMount(Counter, {
      propsData: { initCount: 0 }
    });

    wrapper.vm.inc();
    expect(wrapper).toMatchSnapshot();
  });
});