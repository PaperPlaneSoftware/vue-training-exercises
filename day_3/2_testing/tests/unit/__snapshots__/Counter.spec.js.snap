// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue snapshot testing Snapshots should remain unchanged 1`] = `<div class="counter-container"><button>-</button> <span>1</span> <button>+</button></div>`;
